subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPluginExtension.java

```diff
@@ -92,13 +101,8 @@ public void setReportsDir(File reportsDir) {
     public <T extends Task & JavaForkOptions> void applyTo(final T task) {
         final String taskName = task.getName();
         logger.debug("Applying Jacoco to " + taskName);
-        final JacocoTaskExtension extension = task.getExtensions().create(TASK_EXTENSION_NAME, JacocoTaskExtension.class, agent, task);
-        ((IConventionAware) extension).getConventionMapping().map("destinationFile", new Callable<File>() {
-            @Override
-            public File call() {
-                return project.file(String.valueOf(project.getBuildDir()) + "/jacoco/" + taskName + ".exec");
-            }
-        });
+        final JacocoTaskExtension extension = task.getExtensions().create(TASK_EXTENSION_NAME, JacocoTaskExtension.class, project, agent, task);
+        extension.setDestinationFile(project.file(String.valueOf(project.getBuildDir()) + "/jacoco/" + taskName + ".exec"));
```

Comment:
should be setting using a `Provider<File>` as `project.getBuildDir()` is mutable.